code/modules/tools/tool_archetype.dm

```diff
@@ -0,0 +1,45 @@
+/decl/tool_archetype
+	var/name = "tool"                             // Noun for the tool.
+	var/article = TRUE                            // Boolean value for prefixing 'a' or 'an' to the tool name.
+	var/use_sound = 'sound/items/Deconstruct.ogg' // Sound or list of sounds to play when this tool is used.
+	var/config_sound = 'sound/items/Ratchet.ogg'  // Sound or list of sounds to play when this tool is selected as a variable tool head.
+	var/codex_key
+
+/decl/tool_archetype/proc/can_use_tool(var/atom/tool, var/expend_fuel = 0)
+	return istype(tool) && tool.get_tool_quality(type) > 0
+
+/decl/tool_archetype/proc/handle_pre_interaction(var/mob/user, var/atom/tool, var/expend_fuel = 0)
+	return TRUE
+
+/decl/tool_archetype/proc/handle_post_interaction(var/mob/user, var/atom/tool, var/expend_fuel = 0)
+	return TRUE
+
+/decl/tool_archetype/proc/handle_interaction(var/mob/user, var/atom/tool, var/atom/target, var/expend_fuel = 0)
+	. = istype(user) && istype(target) && can_use_tool(tool, expend_fuel)
+
+/decl/tool_archetype/proc/get_tool_quality_descriptor(var/value)
+
+	if(value < TOOL_QUALITY_WORST)
+		return
+
+	if(value < TOOL_QUALITY_BAD)
+		. = pick("crude", "awful", "dysfunctional")
+	if(value < TOOL_QUALITY_MEDIOCRE)
+		. = "bad"
+	if(value < TOOL_QUALITY_DEFAULT)
+		. = "mediocre"
+	if(value < TOOL_QUALITY_DECENT)
+		. = "adequate"
+	if(value < TOOL_QUALITY_GOOD)
+		. = "decent"
+	else
+		. = pick("excellent", "exquisite", "great")
+			
+	if(.)
+		if(article)
+			. = "[(lowertext(.[1]) in global.vowels_no_u) ? "an" : "a"] [.]"
```

Comment:
The \a macro doesn't work here? Also, I don't think string[n] works (at least on 513); I believe you have to use copytext.

Reply:
Oh, this is because the code previously had 'ugly' as an adjective and `\a ugly` produces `a ugly`. I'll remove it.